subsys/bluetooth/mesh/scheduler_srv.c

```diff
@@ -639,6 +639,12 @@ static int action_set(const struct bt_mesh_model *model, struct bt_mesh_msg_ctx
 		run_scheduler(srv);
 	}
 
+	/* If the next scheduled action is set to "NO_ACTIONS", cancel it. */
+	if ((srv->sch_reg[idx].action == BT_MESH_SCHEDULER_NO_ACTIONS) && (srv->idx == idx)) {
```

Comment:
What will happen if `BT_MESH_SCHEDULER_NO_ACTIONS` came for not ongoing action?

Reply:
Good point, changed it to check whether the bitmap is active for the corresponding index, and to only run the scheduler if the canceled task is first in line.